File: solutions/go/house/1/house.go
package house

import "strings"

type Actor struct {
	name   string
	action string
}

var actors = map[int]Actor{
	1:  {name: "house that Jack built", action: "lay in"},
	2:  {name: "malt", action: "ate"},
	3:  {name: "rat", action: "killed"},
	4:  {name: "cat", action: "worried"},
	5:  {name: "dog", action: "tossed"},
	6:  {name: "cow with the crumpled horn", action: "milked"},
	7:  {name: "maiden all forlorn", action: "kissed"},
	8:  {name: "man all tattered and torn", action: "married"},
	9:  {name: "priest all shaven and shorn", action: "woke"},
	10: {name: "rooster that crowed in the morn", action: "kept"},
	11: {name: "farmer sowing his corn", action: "belonged to"},
	12: {name: "horse and the hound and the horn", action: ""},
}

func Verse(v int) string {
	var sb strings.Builder
	sb.WriteString("This is the ")
	sb.WriteString(actors[v].name)
	for i := v - 1; i > 0; i-- {
		sb.WriteString("\nthat ")
		sb.WriteString(actors[i].action)
		sb.WriteString(" the ")
		sb.WriteString(actors[i].name)
	}
	sb.WriteString(".")
	return sb.String()
}

func Song() string {
	var sb strings.Builder
	for i := 1; i <= len(actors); i++ {
		sb.WriteString(Verse(i))
		if i != len(actors) {
			sb.WriteString("\n\n")
		}
	}
	return sb.String()
}

File: solutions/lua/house/1/house.lua
local house = {}

local parts = {
  'the house that Jack built.',
  'the malt\nthat lay in ',
  'the rat\nthat ate ',
  'the cat\nthat killed ',
  'the dog\nthat worried ',
  'the cow with the crumpled horn\nthat tossed ',
  'the maiden all forlorn\nthat milked ',
  'the man all tattered and torn\nthat kissed ',
  'the priest all shaven and shorn\nthat married ',
  'the rooster that crowed in the morn\nthat woke ',
  'the farmer sowing his corn\nthat kept ',
  'the horse and the hound and the horn\nthat belonged to '
}

house.verse = function(which)
  local verse = ''
  for index = 1, which do
    verse = parts[index] .. verse
  end
  return 'This is ' .. verse
end

house.recite = function()
  local verses = {}
  for index = 1, #parts do
    table.insert(verses, house.verse(index))
  end
  return table.concat(verses, '\n')
end

return house
